astroid/bases.py

```diff
@@ -105,9 +105,13 @@ class Proxy:
     if new instance attributes are created. See the Const class
     """
 
-    _proxied = None  # proxied object may be set by class or by instance
+    _proxied: nodes.ClassDef | nodes.Lambda | Proxy | None = (
+        None  # proxied object may be set by class or by instance
+    )
 
-    def __init__(self, proxied=None):
+    def __init__(
+        self, proxied: nodes.ClassDef | nodes.Lambda | Proxy | None = None
+    ) -> None:
```

Comment:
**Lambda**:
No, I thought it was `FunctionDef` initially but there are multiple tests where `proxied` is `Lambda`. For example:
```python
tests/unittest_inference.py::InferenceTest::test_context_call_for_context_managers
tests/unittest_inference.py::InferenceTest::test_inner_value_redefined_by_subclass
tests/unittest_inference.py::InferenceTest::test_inner_value_redefined_by_subclass_with_mro
tests/unittest_inference.py::InferenceTest::test_lambda_as_methods
tests/unittest_inference.py::InferenceTest::test_subscript_supports__index__
```

**Proxy**:
`proxied` can be `UnboundMethod`, `BoundMethod` but also `List`. Which is a subclass of `(Base)Instance` through `BaseContainer`.
Since I didn't see why one `BaseInstance` would be allowed but another wouldn't I thought it should probably be `BaseInstance | UnboundMethod`. But those both have `Proxy` in common, which is why I settled for that.
`nodes.ClassDef | nodes.Lambda | nodes.List | UnboundMethod | None` would also work with our current tests, but as I said I'm not sure why `nodes.List` is allowed, but other `Instances` aren't.

Reply:
That makes sense. Thanks for the explanation!